src/AutoRest.CSharp/Common/Output/Models/Types/ObjectType.cs

```diff
@@ -33,6 +34,7 @@ protected ObjectType(string defaultNamespace, SourceInputModel? sourceInputModel
         }
 
         public bool IsStruct => ExistingType?.IsValueType ?? false;
+        protected override TypeKind TypeKind => IsStruct ? TypeKind.Struct : TypeKind.Class;
```

Comment:
Previously this override is in `SchemaObjectType` which is the reason this feature is not working properly for `ModelTypeProvider`.
Now we move the implementation here, so that `SchemaObjectType` and `ModelTypeProvider` could get this implementation.